src/declarativewebcontainer.cpp

```diff
@@ -655,8 +638,12 @@ void DeclarativeWebContainer::releasePage(int tabId, bool virtualize)
     if (m_webPages) {
         m_webPages->release(tabId, virtualize);
         // Successfully destroyed. Emit relevant property changes.
-        if (!m_webPage) {
-            m_tabId = 0;
+        if (!m_webPage || m_model->count() == 0) {
+            if (m_tabId != 0) {
+                m_tabId = 0;
+                emit tabIdChanged();
```

Comment:
Previously sequence emitted this signal as a last signal. Updating data over here is fine.

So, when tabId has changed all data always matches to the tabId, no wrong title/url etc.